SIT.Manager.Avalonia/Services/LocalizationService.cs

```diff
@@ -0,0 +1,119 @@
+        {
+            if (resourceInclude == null || string.IsNullOrEmpty(currentLanguage) || currentLanguage != _configService.Config.CurrentLanguageSelected)
+            {
+                try
+                {
+                    resourceInclude = CreateResourceLocalization($"avares://SIT.Manager.Avalonia/Localization/{_configService.Config.CurrentLanguageSelected}.axaml");
+                }
+                catch // If there was an issue loading current Culture language, we will default by English.
+                {
+                    resourceInclude = CreateResourceLocalization($"avares://SIT.Manager.Avalonia/Localization/en-US.axaml");
+                }
+            }
+
+            string result = "not found";
+            if (resourceInclude.TryGetResource(key, null, out object? translation))
+            {
+                if (translation != null)
+                {
+                    result = (string)translation;
+                    for (int i = 0; i < replaces.Length; i++)
+                    {
+                        result = result.Replace($"%{i + 1}", replaces[i]);
+                    }
+                }
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Function that loads the Available Localizations when program starts.
+        /// </summary>
+        public List<CultureInfo> GetAvailableLocalizations()
+        {
+            List<CultureInfo> result = []; // Initialize the list properly
+            var assembly = typeof(LocalizationService).Assembly;
+            string folderName = string.Format("{0}.Localization", assembly.GetName().Name);
+            result = assembly.GetManifestResourceNames()
+                .Where(r => r.StartsWith(folderName) && r.EndsWith(".axaml"))
+                .Select(r =>
+                {
+                    string[] parts = r.Split('.');
+                    string languageCode = parts[^2];
+                    try
+                    {
+                        CultureInfo cultureInfo = new(languageCode);
+                        return cultureInfo;
+                    }
+                    catch (CultureNotFoundException)
+                    {
+                        CultureInfo cultureInfo = new("en-US");
+                        return cultureInfo;
+                    }
+                })
+                .Where(cultureInfo => cultureInfo != null)
+                .ToList();
+            return result;
+        }
+
+        /// <summary>
+        /// Creates a Resource that will load Localization later on.
+        /// </summary>
+        /// <returns>Resource with Localization</returns>
+        private ResourceInclude CreateResourceLocalization(string url)
+        {
+            var self = new Uri("resm:Styles?assembly=SIT.Manager.Avalonia");
+            return new ResourceInclude(self)
+            {
+                Source = new Uri(url)
+            };
+        }
```

Comment:
I don't like this approach. We're repeating ourselves massively here. We already pull values such as `SIT.Manager.Avalonia.Localization.en-US.axaml` from the first half of the LINQ expression, only to extract the country code, convert it into a CultureInfo to the reconstruct that exact path again later to create the resource